File: app/src/main/java/life/homail/calculator/NumberBtnEventHandler.java
package life.homail.calculator;
import android.view.View;
public class NumberBtnEventHandler implements View.OnClickListener{
    private CalculatorMain calculatorMain;
    protected NumberBtnEventHandler(CalculatorMain calculatorMain){
        this.calculatorMain=calculatorMain;
    }

    @Override
    public void onClick(View view){
        this.findTheBtnAndPerformActionsAccordingly(view);
        this.calculatorMain.soundHandlerAndPlayer.playSound();
    }

    private void findTheBtnAndPerformActionsAccordingly(View view){
        if (view.getId()==R.id.button0){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.button0.getText());
        } else if (view.getId()==R.id.button1){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.button1.getText());
        } else if (view.getId()==R.id.button2){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.button2.getText());
        } else if (view.getId()==R.id.button3){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.button3.getText());
        } else if (view.getId()==R.id.button4){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.button4.getText());
        } else if (view.getId()==R.id.button5){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.button5.getText());
        } else if (view.getId()==R.id.button6){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.button6.getText());
        } else if (view.getId()==R.id.button7){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.button7.getText());
        } else if (view.getId()==R.id.button8){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.button8.getText());
        } else if (view.getId()==R.id.button9){
            this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(this.calculatorMain.calculatorViews.button9.getText());
        }
    }
}

File: app/src/main/java/life/homail/calculator/NumberBtnEventHandler.kt
package life.homail.calculator
import android.view.View
import android.widget.Button
class NumberBtnEventHandler(private val calculatorMain: CalculatorMain) :  View.OnClickListener{
    private var clickedButton: Button?=null
    override fun onClick(view:View){
        this.clickedButton=this.findTheButtonClicked(view)
        this.setTextOnScreen(this.clickedButton)
    }
    private fun setTextOnScreen(clickedButton: Button?){
        this.calculatorMain.calculatorViews.textFieldWhereUserEnters.append(clickedButton?.text.toString())
    }
    private fun findTheButtonClicked(view:View):Button?{
        when (view.id) {
            R.id.button0 -> return this.calculatorMain.calculatorViews.button0;
            R.id.button1 -> return this.calculatorMain.calculatorViews.button1;
            R.id.button2 -> return this.calculatorMain.calculatorViews.button2;
            R.id.button3 -> return this.calculatorMain.calculatorViews.button3;
            R.id.button4 -> return this.calculatorMain.calculatorViews.button4;
            R.id.button5 -> return this.calculatorMain.calculatorViews.button5;
            R.id.button6 -> return this.calculatorMain.calculatorViews.button6;
            R.id.button7 -> return this.calculatorMain.calculatorViews.button7;
            R.id.button8 -> return this.calculatorMain.calculatorViews.button8;
            R.id.button9 -> return this.calculatorMain.calculatorViews.button9;
        }
        return null;
    }
}